### PR TITLE
Add plugin for datasources

### PR DESCRIPTION
**Pull request by tdraier**
## Description

Add a poke plugin for dataSource

## Risk

none

## Deploy Plan

deploy front

### CODE REVIEW

File: front/lib/api/poke/plugins/data_sources/operations.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { assertNever, ConnectorsAPI, Err, Ok } from "@dust-tt/types";
+
+import config from "@app/lib/api/config";
+import { createPlugin } from "@app/lib/api/poke/types";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import logger from "@app/logger/logger";
+
+const OPERATIONS = ["STOP", "PAUSE", "UNPAUSE", "RESUME", "SYNC"] as const;
+
+const doOperation = (op: (typeof OPERATIONS)[number], connectorId: string) => {
```
Comment (flvndvd) [inline front/lib/api/poke/plugins/data_sources/operations.ts:10]: nit: Let's extract (typeof OPERATIONS)[number] to a proper type.

File: front/lib/api/poke/plugins/data_sources/operations.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      op: {
+        type: "enum",
+        label: "Operation",
+        description: "Select operation to execute",
+        values: OPERATIONS,
+      },
+    },
+  },
+  async (auth, dataSourceId, args) => {
+    if (!dataSourceId) {
```
Comment (flvndvd) [inline front/lib/api/poke/plugins/data_sources/operations.ts:48]: This should never happen, right? I'm pretty sure the type enforce this!

File: front/lib/api/poke/plugins/data_sources/operations.ts (hunk 1 above)
Comment (flvndvd) [inline front/lib/api/poke/plugins/data_sources/operations.ts:10]: IIRC, our CLI is exposed through the connector API as well, don't we want to use it?

**Comment by flvndvd** [comment]
🔥 🔥 🔥 🔥

File: front/lib/api/poke/plugins/data_sources/operations.ts (hunk 2 above)
Comment (tdraier) [reply front/lib/api/poke/plugins/data_sources/operations.ts:48]: the type is `resourceId: string | undefined` , so yes, could happen

File: front/lib/api/poke/plugins/data_sources/operations.ts (hunk 1 above)
Comment (tdraier) [reply front/lib/api/poke/plugins/data_sources/operations.ts]: The /admin endpoint would be interesting to add specific plugins per provider type, but commands have to be whitelisted to be callable via api , and some commands require more parameters - let's keep that for later

**Review by flvndvd — APPROVED** [review]
Thank you 🙏

**Pull request merged**